inc/impact.class.php

```diff
@@ -0,0 +1,1091 @@
+         );
+      }
+
+      return self::createTabEntry(
+         _n('Impact ', 'Impacts ', Session::getPluralNumber()),
+         $total
+      );
+   }
+
+   public static function displayTabContentForItem(
+      CommonGLPI $item,
+      $tabnum = 1,
+      $withtemplate = 0
+   ) {
+      global $CFG_GLPI;
+
+      $class = get_class($item);
+
+      // Only enabled for CommonDBTM
+      if (!is_a($item, "CommonDBTM")) {
+         throw new InvalidArgumentException(
+            "Argument \$item ($class) must be a CommonDBTM)."
+         );
+      }
+
+      $ID = $item->getID();
+
+      // Don't show the impact analysis on new object
+      if ($item->isNewID($ID)) {
+         return false;
+      }
+
+      // Check READ rights
+      $itemtype = $item->getType();
+      if (!$itemtype::canView()) {
+         return false;
+      }
+
+      // For an ITIL object, load the first linked element by default
+      if (is_a($item, "CommonITILObject")) {
+         $linked_items = $item->getLinkedItems();
+
+         // Search for a valid linked item of this ITILObject
+         $found = false;
+         foreach ($linked_items as $linked_item) {
+            $class = $linked_item['itemtype'];
+            if (isset($CFG_GLPI['impact_asset_types'][$class])) {
+               self::printAssetSelectionForm($linked_items);
+               $found = true;
+               $item = new $class;
+               $item->getFromDB($linked_item['items_id']);
+               break;
+            }
+         }
+
+         // No valid linked item were found, tab shouldn't be visible
+         if (!$found) {
+            return true;
+         }
+      }
+
+      // Show graph if the impact analysis is enable for $class
+      if (isset($CFG_GLPI['impact_asset_types'][$class])) {
+         self::loadLibs();
+         self::prepareImpactNetwork($item);
+         self::buildNetwork($item);
+      }
+
+      return true;
+   }
+
+   /**
+    * Load the cytoscape and spectrum-colorpicker librairies
+    *
+    * @since 9.5
+    */
+   public static function loadLibs() {
+      echo Html::css('public/lib/spectrum-colorpicker.css');
+      echo Html::script("public/lib/spectrum-colorpicker.js");
+      echo Html::css('public/lib/cytoscape.css');
+      echo Html::script("public/lib/cytoscape.js");
+   }
+
+   /**
+    * Print the asset selection form used in the impact tab of ITIL objects
+    *
+    * @param array $items
+    *
+    * @since 9.5
+    */
+   public static function printAssetSelectionForm(array $items) {
+      global $CFG_GLPI;
+
+      // Dropdown values
+      $values = [];
+
+      // Add a value in the dropdown for each items, grouped by type
+      foreach ($items as $item) {
+         if (isset($CFG_GLPI['impact_asset_types'][$item['itemtype']])) {
+            // Add itemtype group if it doesn't exist in the dropdown yet
+            $itemtype_label = __($item['itemtype']);
```

Comment:
I'm not sure all itemtypes do have a translation. Better to use `getTypeName()` I guess

Reply:
Fixed in latest commit.